DynamicQueryEngine/DynamicQueryEngine.cs

```diff
@@ -140,13 +143,13 @@ static DynamicQueryEngine()
 					fieldToParameter.Add(field, newParameter);
 				}
 			}
-			if(fieldNames.Count<=0)
+			if (fieldNames.Count <= 0)
 			{
-				if(hasIdentity)
+				if (hasIdentity)
 				{
 					// a table with just 1 identity field, use a special case query: INSERT INTO table values ()
-					fieldNames.Clear();
-					valueFragments.Clear();
+					//fieldNames.Clear();
```

Comment:
This is the reason;

'QueryFragments' does not contain a definition for 'Clear' and no extension method 'Clear' accepting a first argument of type 'QueryFragments' could be found 

Not sure what the alternative should be? I just looked at the ASE repo and found that no 'Clear'-method was called there

Reply:
These clear calls are not needed. It builds a stack of objects in which strings are stored which is then flattened into a single buffer, to avoid unnecessary concats and inserts. Then this stack is discarded, so clear isn't needed.